### PR TITLE
Pva/evsrestapi 483 client sdk typescript openapi

### PR DESCRIPTION
**Pull request by peter-va**
https://tracker.nci.nih.gov/browse/EVSRESTAPI-481

typescript openapi implementation. implemented tests match other examples, as does readme structure

### CODE REVIEW

File: typescript-examples/servers.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * @param endpoint the endpoint to be queried on the server
+     * @param httpMethod httpMethod to be used
+     *
+     */
+    public makeRequestContext(endpoint: string, httpMethod: HttpMethod): RequestContext {
+        return new RequestContext(this.getUrl() + endpoint, httpMethod);
+    }
+}
+
+export const server1 = new ServerConfiguration<{  }>(baseUrl, {  })
```
Comment (gaaliciA1990) [inline typescript-examples/servers.ts:60]: This is a change you made to allow us to set the baseURL, which is great, but that means this file should not be overridden when regenerating the code. Make sure this is documented in the `doc/code-gen/BUILDME.md`

**Review by gaaliciA1990 — CHANGES_REQUESTED** [review]
Overall, this is great. We are missing some documentation that is necessary for regenerating the code. See my feedback in the specified comments. 
Add Typescript to the `BUILDME.md` file for how to regenerate the code, what to copy over and what not to override (like `servers.ts`). I wasn't able to identify all of the classes that had been manually modified to meet our needs, but if there are more, we want them explicitly stated and how to handle new generated code for those class.

File: typescript-examples/servers.ts (hunk 1 above)
Comment (peter-va) [reply typescript-examples/servers.ts:60]: added the information to buildme.md

**Review by gaaliciA1990 — APPROVED** [review]
Everything looks good. Approved and ready to merge with develop

**Pull request merged**